rollup.config.js

```diff
@@ -0,0 +1,22 @@
+import commonjs from "@rollup/plugin-commonjs";
+import resolve  from "@rollup/plugin-node-resolve";
+import json     from "@rollup/plugin-json";
+import fs       from "fs";
+
+
+// the espree package version is exported from the module (i.e., import { version } from "espree")
+// read version from the package.json metadata and write it into lib/version.js at build time,
+// since esm cannot import json by default
+const pkg = JSON.parse(fs.readFileSync("./package.json", "utf8"));
+fs.writeFileSync("lib/version.js", `const version = "${pkg.version}";\n\nexport default version;\n`);
```

Comment:
Is this a common practice? We should revisit our [release steps](https://github.com/eslint/eslint-release/blob/master/lib/release-ops.js#L344) to make sure that `package.json` has been updated before running this code.

Reply:
I don't know if it's common practice or not. Personally I don't see why the version number even needs to be explicitly exported from the module at all, but this seemed like an easy way to accomplish that without adding even more plugins and tooling.

Reply:
I think this is fine for now. In the future, Node.js will support JSON imports and we'll be able to remove this.

Reply:
We should modify our release process.

Current steps in [generateRelease](https://github.com/eslint/eslint-release/blob/c1391a40b633315ee7b79cffad7043491f9e44f9/lib/release-ops.js#L344) are:

1. `npm test`
2. `npm version` to update `package.json`

But we should have this order now:

1. `npm version` to update `package.json`
2. `npm run rollup` to make `lib/version.js` and `dist/espree.cjs`.
3. `npm test`

Maybe we could reorder the steps in `generateRelease` (which is used for all packages, not just `espree`) to calculate the new version and update `package.json` before running `npm test`, and also update the `test` script to run `rollup` before the tests.

Reply:
In the current esm PR running `npm run rollup` implicitly updates `version.js`.

I did it this way because I don't know of any use cases where one would want to update version and _not_ re-package the commonjs build in `dist/`.